### PR TITLE
Custom palette

### PR DESCRIPTION
**Pull request by suyashmahar**
This PR is for testing purposes please ignore. Just in case you are getting [bored](https://suyashmahar.me/404)

### CODE REVIEW

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 1]
```diff
@@ -1,102 +1,180 @@
...
-    PuOr(0x7f3b08, 0xb35806, 0xe08214, 0xfdb863, 0xfee0b6, 0xf7f7f7, 0xd8daeb, 0xb2abd2, 0x8073ac),
-    BrBG(0x543005, 0x8c510a, 0xbf812d, 0xdfc27d, 0xf6e8c3, 0xf5f5f5, 0xc7eae5, 0x80cdc1, 0x35978f),
-    PRGn(0x40004b, 0x762a83, 0x9970ab, 0xc2a5cf, 0xe7d4e8, 0xf7f7f7, 0xd9f0d3, 0xa6dba0, 0x5aae61),
-    PiYG(0x8e0152, 0xc51b7d, 0xde77ae, 0xf1b6da, 0xfde0ef, 0xf7f7f7, 0xe6f5d0, 0xb8e186, 0x7fbc41),
-    RdBu(0x67001f, 0xb2182b, 0xd6604d, 0xf4a582, 0xfddbc7, 0xf7f7f7, 0xd1e5f0, 0x92c5de, 0x4393c3),
-    RdGy(0x67001f, 0xb2182b, 0xd6604d, 0xf4a582, 0xfddbc7, 0xffffff, 0xe0e0e0, 0xbababa, 0x878787),
-    RdYlBu(0xa50026, 0xd73027, 0xf46d43, 0xfdae61, 0xfee090, 0xffffbf, 0xe0f3f8, 0xabd9e9, 0x74add1),
-    Spectral(0x9e0142, 0xd53e4f, 0xf46d43, 0xfdae61, 0xfee08b, 0xffffbf, 0xe6f598, 0xabdda4, 0x66c2a5),
-    RdYlGn(0xa50026, 0xd73027, 0xf46d43, 0xfdae61, 0xfee08b, 0xffffbf, 0xd9ef8b, 0xa6d96a, 0x66bd63);
+    private int colors[];
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: Array brackets at illegal position\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 2]
```diff
@@ -1,102 +1,180 @@
...
-    private final int c6;
-    private final int c7;
-    private final int c8;
+    /**
+     * Return palette object corresponding to supplied value of paletteIndex, palette is constructed
+     * from a predefined set of colors
+     * @param palleteIndex Index of palette to return
+     * @return Palette object generated from predefined set of colors
+     */
+    public static Palette getPalette(int palleteIndex){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: NCSS for this method is 60 \(max allowed is 50\)\.

Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 3]
```diff
@@ -1,102 +1,180 @@
...
+        switch (palleteIndex){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 4]
```diff
@@ -1,102 +1,180 @@
...
-        this.c5 = c5;
-        this.c6 = c6;
-        this.c7 = c7;
-        this.c8 = c8;
+    /**
+     * Returns index of palette object passed from list of palettes predefined in Palette
+     * @param palette Object for finding index
+     * @return Index from predefined pallete or -1 if not found
+     */
+    public static int indexOf(Palette palette){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 5]
```diff
@@ -1,102 +1,180 @@
...
+        int pos = -1;
+        for (int i = 0; i < Palette.DEFAULT_PALETTE_COUNT; i++){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 6]
```diff
@@ -1,102 +1,180 @@
...
+            if (Palette.getPalette(i) == palette) {
+                pos = i;
+            }
+        }
+        return  pos;
+    }
+
+    public Palette(int c0, int c1, int c2, int c3, int c4, int c5, int c6, int c7, int c8) {
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: More than 7 parameters \(found 9\)\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 7]
```diff
@@ -1,102 +1,180 @@
...
+        colors[2] = c2;
+        colors[3] = c3;
+        colors[4] = c4;
+        colors[5] = c5;
+        colors[6] = c6;
+        colors[7] = c7;
+        colors[8] = c8;
     }
 
+    public Palette(int colors[]) {
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: \[Checkstyle\] ERROR: Array brackets at illegal position\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java (hunk 2 above)
Comment (suyashmahar) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java]: It is not valid here

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 8]
```diff
@@ -1,102 +1,191 @@
...
-                return c7;
-            case 8:
-                return c8;
+    /**
+     * Return palette object corresponding to supplied value of paletteIndex, palette is constructed
+     * from a predefined set of colors
+     * @param paletteIndex Index of palette to return
+     * @return Palette object generated from predefined set of colors
+     */
+    public static Palette getPalette(int paletteIndex) {
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java:64]: \[Checkstyle\] ERROR: NCSS for this method is 60 \(max allowed is 50\)\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 9]
```diff
@@ -1,102 +1,191 @@
...
+        switch (paletteIndex){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java:65]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 10]
```diff
@@ -1,102 +1,191 @@
...
             }
         }
         return  pos;
     }
 
-    public static void main(String[] args) {
-        if (args.length != 1) {
-            System.err.println("Usage: java Palette <colors>");
-            System.exit(-1);
+    public Palette(int c0, int c1, int c2, int c3, int c4, int c5, int c6, int c7, int c8) {
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java:142]: \[Checkstyle\] ERROR: More than 7 parameters \(found 9\)\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 11]
```diff
@@ -1,102 +1,191 @@
...
+        colors[2] = c2;
+        colors[3] = c3;
+        colors[4] = c4;
+        colors[5] = c5;
+        colors[6] = c6;
+        colors[7] = c7;
+        colors[8] = c8;
+    }
+
+    public Palette(int colors[]) {
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java:155]: \[Checkstyle\] ERROR: Array brackets at illegal position\.

File: trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java [hunk 12]
```diff
@@ -1,102 +1,191 @@
...
+        this.colors = colors;
+    }
+
+    /**
+     * Returns color corresponding to index passed from the set of color for a palette
+     * @param index Index of color in set of color for current palette object
+     * @return color as int without alpha channel
+     */
+    public int getColor(int index) {
+        switch (index){
```
Comment (sputnikci) [inline trianglify/src/main/java/com/sdsmdg/kd/trianglify/models/Palette.java:168]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

**Pull request closed without merging**